### DISCUSSION TITLE
Is there any method to remove or ignore the footer and header text?

### DISCUSSION THREAD

**Discussion by hemilparmar** [pymupdf/PyMuPDF#1930] [Q&A]
can we remove header and footer from pdf while extracting the text?

**Comment by JorjMcKie** [comment]
No, because PDF doesn't know what these things are. It's all just text, and you know better than anyone what to exclude eventually.

**Reply by JorjMcKie** [reply]
For example, if you know the position of headers / footers on a page, then build a sub-rectangle of the page rectangle and limit extraction to it.
```python
# assume 'top' is the top coordinate of relevant text
# assume 'btm' is the bottom coordinate of relevant text
clip = page.rect
clip.y0 = top
clip.y1 = btm
# all text extraction variants accept the clip parameter:
text = page.get_text(clip=clip)
```

**Reply by hemilparmar** [reply]
As you said for parser , it is all text. so is there any way to know the header/footer location using fitz.?

**Reply by JorjMcKie** [reply]
For every text on the page, you can determine its position. The `Page.get_text()` method has several ways to get this information in varying levels of detail:
* get_text("blocks") - delivers a list of text blocks, togehter with the rectangle of the block
* get_text("words") - similar list, but on word level
* get_text("dict") - a dictionary with position info for each block, line or text "span"
* get_text("rawdict") - a similar dictionary, but with one more level down to **_each character_**

Please see the documentation for more details.

**Reply by binhpageai** [reply]
can you explain how pymupdf group characters/words to blocks?

**Comment by shiu886** [comment]
There might be a separating line between the text and footer. Usually footer would be in a smaller font size.

**Reply by JorjMcKie** [reply]
@shiu886 - you are right: headers / footers often are written with other font properties than the regular text like smaller, or even a different font.

**Comment by JorjMcKie** [comment]
Please have a look at the new AI-powered version 0.2.2 of [PyMuPDF4LLM](https://github.com/pymupdf/pymupdf4llm).
It supports PyMuPDF's layout detection package [PyMuPDF-Layout)[https://pypi.org/project/pymupdf-layout/).

You can find a feature comparison [here](https://github.com/pymupdf/pymupdf4llm/discussions/327).

PyMuPDF4LLM's extraction methods `to_markdown()` and `to_text()` allow suppression of page headers and foots - among a plethora of other features.
